dashboard/legacy/middleware/channels_api.rb

```diff
@@ -403,4 +400,18 @@ def project_validator?
   def verified_teacher?
     has_permission?("authorized_teacher")
   end
+
+  def check_can_publish(channel_id, project_type)
+    not_authorized unless owns_channel?(channel_id)
+    bad_request unless ALL_PUBLISHABLE_PROJECT_TYPES.include?(project_type)
+    forbidden if sharing_disabled? && CONDITIONALLY_PUBLISHABLE_PROJECT_TYPES.include?(project_type)
+    return unless RESTRICTED_PUBLISH_PROJECT_TYPES.include?(project_type)
+
+    # Check for restricted share mode. If we are in restricted share mode, it means we cannot publish.
+    source_data = SourceBucket.new.get(channel_id, "main.json")
+    return unless source_data && source_data[:body] && source_data[:body].respond_to?(:string)
+    source_body = source_data[:body].string
+    project_src = ProjectSourceJson.new(source_body)
+    forbidden if project_src.in_restricted_share_mode?
```

Comment:
Anywhere we can encapsulate this check so it can be shared by channels_api and projects_controller?

Reply:
good call out, I'll try to find a place to put this